src/Altinn.Notifications.Core/Services/ContactPointService.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Altinn.Notifications.Core.Integrations;
+using Altinn.Notifications.Core.Models;
+using Altinn.Notifications.Core.Models.Address;
+using Altinn.Notifications.Core.Models.ContactPoints;
+using Altinn.Notifications.Core.Services.Interfaces;
+
+namespace Altinn.Notifications.Core.Services
+{
+    /// <summary>
+    /// Implementation of the <see cref="IContactPointService"/> using Altinn platform services to lookup contact points
+    /// </summary>
+    public class ContactPointService : IContactPointService
+    {
+        private readonly IProfileClient _profileClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ContactPointService"/> class.
+        /// </summary>
+        public ContactPointService(IProfileClient profile)
+        {
+            _profileClient = profile;
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<Recipient>> GetEmailContactPoints(List<Recipient> recipients)
+        {
+            return await AugmentRecipients(
+                recipients,
+                (recipient, userContactPoints) =>
+                {
+                    if (userContactPoints.IsReserved)
+                    {
+                        recipient.IsReserved = userContactPoints.IsReserved;
+                    }
```

Comment:
Looks like these lines could be moved into the `AugmentRecipients` method.

Reply:
Can also be simplified?
```suggestion
                    recipient.IsReserved = userContactPoints.IsReserved;
```